Serialiser_Engine/Convert/Json.cs

```diff
@@ -77,16 +77,25 @@ public static object FromJson(string json)
             else if (json.StartsWith("["))
             {
                 Reflection.Compute.RecordNote("The string provided represents a top-level Json Array instead of a Json object.\nDeserializing to a CustomObject.");
+
                 BsonArray array = MongoDB.Bson.Serialization.BsonSerializer.Deserialize<BsonArray>(json);
 
-                return new BH.oM.Base.CustomObject
+                List<object> objects = new List<object>();
+
+                for (int i=0; i<array.Count; i++)
+                {
+                    object obj = FromJson(array[i].ToString());
+                    objects.Add(obj);
+                }
+                return new BH.oM.Base.CustomObject()
                 {
                     CustomData = new Dictionary<string, object>()
                     {
-                        { "Objects", array.Select(b => Convert.FromBson(b.AsBsonDocument)).ToList() }
+                        {"Objects", objects}
```

Comment:
Thanks @enarhi for spotting it.
I suppose we can just change this line if that's the issue, no?
```suggestion
                        { "Objects", array.Select(b => Convert.FromJson(b.ToString())).ToList() }
```